src/com/facebook/buck/apple/PrebuiltAppleFramework.java

```diff
@@ -0,0 +1,279 @@
+import com.facebook.buck.rules.AddToRuleKey;
+import com.facebook.buck.rules.BuildContext;
+import com.facebook.buck.rules.BuildRuleParams;
+import com.facebook.buck.rules.BuildRuleResolver;
+import com.facebook.buck.rules.BuildTargetSourcePath;
+import com.facebook.buck.rules.BuildableContext;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.SourcePathResolver;
+import com.facebook.buck.rules.args.Arg;
+import com.facebook.buck.rules.args.StringArg;
+import com.facebook.buck.rules.coercer.FrameworkPath;
+import com.facebook.buck.step.Step;
+import com.facebook.buck.step.fs.CopyStep;
+import com.facebook.buck.step.fs.MkdirStep;
+import com.facebook.buck.step.fs.RmStep;
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class PrebuiltAppleFramework
+    extends AbstractBuildRule
+    implements AbstractCxxLibrary, HasOutputName {
+
+  @AddToRuleKey(stringify = true)
+  private final Path out;
+
+  private final BuildRuleParams params;
+  private final BuildRuleResolver ruleResolver;
+  private final SourcePath frameworkPath;
+  private final String frameworkName;
+  private final Function<? super CxxPlatform, ImmutableList<String>> exportedLinkerFlags;
+  private final ImmutableSet<FrameworkPath> frameworks;
+  private final Optional<Pattern> supportedPlatformsRegex;
+
+  private final Map<Pair<Flavor, Linker.LinkableDepType>, NativeLinkableInput> nativeLinkableCache =
+      new HashMap<>();
+
+  private final LoadingCache<
+      CxxPreprocessables.CxxPreprocessorInputCacheKey,
+      ImmutableMap<BuildTarget, CxxPreprocessorInput>
+      > transitiveCxxPreprocessorInputCache =
+      CxxPreprocessables.getTransitiveCxxPreprocessorInputCache(this);
+
+  public PrebuiltAppleFramework(
+      BuildRuleParams params,
+      BuildRuleResolver ruleResolver,
+      SourcePathResolver pathResolver,
+      SourcePath frameworkPath,
+      ImmutableSet<FrameworkPath> frameworks,
+      Optional<Pattern> supportedPlatformsRegex,
+      Function<? super CxxPlatform, ImmutableList<String>> exportedLinkerFlags) {
+    super(params, pathResolver);
+    this.params = params;
+    this.frameworkPath = frameworkPath;
+    this.ruleResolver = ruleResolver;
+    this.exportedLinkerFlags = exportedLinkerFlags;
+    this.frameworks = frameworks;
+    this.supportedPlatformsRegex = supportedPlatformsRegex;
+
+    BuildTarget target = params.getBuildTarget();
+    this.frameworkName = pathResolver.getAbsolutePath(frameworkPath).getFileName().toString();
+    this.out =
```

Comment:
Easier to say 

`BuildTargets.getGenPath( getProjectFilesystem(), target, "%s")`